src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2791,6 +2792,50 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
             return objectCreation;
         }
 
+        public override IOperation VisitDeconstructionAssignment(IDeconstructionAssignmentOperation operation, int? captureIdForResult)
+        {
+            // If the assignment target is a tuple, we want to decompose the tuple and push each element back onto the stack, so that if the value
+            // has control flow the individual elements are captured. Then we can recompose the tuple after operation.Value has been visited.
+            // We do this to keep the graph sane, so that users don't have to track a tuple captured via flow control when it's not really
+            // the tuple that's been captured, it's the operands to the tuple.
+            ITupleOperation tupleTarget = null;
+            if (operation.Target.Kind == OperationKind.Tuple)
+            {
+                tupleTarget = (ITupleOperation)operation.Target;
+                PushArray(tupleTarget.Elements);
+            }
+            else
+            {
+                // This is an error case, so we don't try to do any kind of simplification for it
```

Comment:
>// This is an error case, [](start = 16, length = 25)

I don't think this is accurate, this could be a DeclarationExpression in success scenario.